test/integration/traces_test.go

```diff
@@ -135,11 +142,24 @@ func testHTTPTracesBadTraceID(t *testing.T) {
 		"invalid-trace-id3": "00-5fe865607Ra112abd799ea8108c38bcb-4c59e9a913c480a3-01",
 		"invalid-trace-id4": "00-0x5fe865607da112abd799ea8108c3cb-4c59e9a913c480a3-01",
 		"invalid-trace-id5": "00-5FE865607DA112ABD799EA8108C38BCB-4c59e9a913c480a3-01",
+		// For parent/flags test, traceID portion must be different each time
+		"invalid-parent-id1": "00-11111111111111111111111111111111-Zc59e9a913c480a3-01",
```

Comment:
This is really a great validation test, but I would honestly turn these tests here into a unit test on the code here (with some moved parsing logic from spanner.go):
https://github.com/grafana/beyla/pull/258/files#diff-5511390ee7e0e2becd168d5dcc0ea39dc0a0719a1ffdfda2d91c319617271273R283

We can easily extract that logic into a standalone function and verify that it parses correctly, checking for say 0 length traces, traceparents that have completely bogus data (e.g. "hello") and these edge cases below. The reason why the unit tests might be better choice is that it's easy to tell what part of the code isn't working, and it's a lot quicker to run. If we had a failure for the integration test with some of these formats, we'd technically need to write a test to debug the specific string.

Reply:
Thanks for the explanation.  I've added a unit test to see what fails validation.